test/Dialect/MSFT/location.mlir

```diff
@@ -11,8 +27,7 @@ msft.module @leaf {} () -> () {
   // LOWER: hw.instance "module" sym @module @Foo() -> ()
   // LOWER-NOT: #msft.switch.inst
   msft.instance @module @Foo() {
-    "loc:memBank2" = #msft.switch.inst< @shallow["leaf"]=#msft.physloc<M20K, 8, 19, 1>,
-                                        @deeper["branch","leaf"]=#msft.physloc<M20K, 15, 9, 3> >
+    circt.globalRef = [#hw.globalNameRef<@ref1>, #hw.globalNameRef<@ref2>], inner_sym = "module"
```

Comment:
Should we change the symbol attribute on msft.instance and compreg to `inner_sym`?

Reply:
For compreg, there is already an inner symbol. We could rename it to `inner_sym` so it works with all the other stuff that uses `inner_sym`. For `msft.instance`, I'm not sure if it matters, but it is using `sym_name` because it implements the `Symbol` trait. I'm not sure if that is useful.. if not, we can remove `Symbol` and rename the attribute.